### PR TITLE
Dev container

### PR DESCRIPTION
**Pull request by Tom-Latham**
##### SUMMARY

Updated devcontainer to use a public image for building

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

DevContainer

##### ADDITIONAL INFORMATION

Dev container now uses a public python image and installs Ansible in the post-setup script. Also updated the default directory when opening a new terminal in the container to be the root of this repo

```paste below

```

### CODE REVIEW

**Review by andrewhughes101 — APPROVED** [review]
Looks good. Think at some point we should still make this less opinionated by removing the ruff support and pyproject toml, but for now this is a good starting point

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		"ghcr.io/devcontainers/features/git:1": {},
+		"ghcr.io/devcontainers/features/common-utils:2": {}
+	},
+	"privileged": true,
+	"remoteUser": "root",
+	"workspaceMount": "source=${localWorkspaceFolder},target=/workspace/collections/ansible_collections/ibm/ibm_zos_cics,type=bind",
+	"workspaceFolder": "/workspace/collections",
+    "postAttachCommand": "bash /workspace/collections/ansible_collections/ibm/ibm_zos_cics/.devcontainer/setup.sh",
+	"postCreateCommand": {
+        "gitMarkSafe": "git config --global --add safe.directory /workspaces/collections/ansible_collections/ibm/ibm_zos_cics"
```
Comment (andrewhughes101) [inline .devcontainer/devcontainer.json:14]:
```suggestion
	"workspaceMount": "source=${localWorkspaceFolder},target=/workspace/collections/ansible_collections/ibm/ibm_zos_cics,type=bind",
	"workspaceFolder": "/workspace/collections",
	"postAttachCommand": "bash /workspace/collections/ansible_collections/ibm/ibm_zos_cics/.devcontainer/setup.sh",
	"postCreateCommand": {
        "gitMarkSafe": "git config --global --add safe.directory /workspaces/collections/ansible_collections/ibm/ibm_zos_cics"
```

**Review by andrewhughes101 — APPROVED** [review]
Minor whitespace fixup

**Comment by andrewhughes101** [comment]
Re-launch CI

**Comment by Tom-Latham** [comment]
Re-launch CI

**Pull request merged**